packages/utils/src/artifact.ts

```diff
@@ -7,7 +7,7 @@ export interface AbiInput {
 
 export interface AbiFunction {
   name: string;
-  covenant: boolean;
+  covenant?: boolean;
```

Comment:
Is this the one place that the old covenant interface has to remain?

Reply:
Yes, I decided to keep the CashScript SDK compatible with the old way of doing covenants so that the new version of the SDK can still be used with older versions of the language/compiler. So that's why `covenant` is still a *potential* option in the artifact format.

Old covenants will be deprecated though so at some point that compatibility will also be removed from the SDK.